resharper/src/resharper-unity/CSharp/Daemon/Errors/CSharpErrors.xml

```diff
@@ -180,7 +192,34 @@
     <Range>EqualityExpression.GetHighlightingRange()</Range>
     <Behavour overlapResolvePolicy="WARNING" />
   </Warning>
+  
+  <Warning name="UseExplicitTypeInsteadOfStringUsing" configurableSeverity="Unity.UseExplicitTypeInsteadOfStringUsing">
+    <Parameter type="IInvocationExpression" name="invocationMethod" />
+    <Parameter type="string" name="methodName" isValid="skip"/>
+    <Parameter type="string" name="stringLiteral" isValid="skip"/>
+    <Parameter type="IExpression" name="argument" />
+    <Parameter type="ITypeElement[]" name="availableTypes" isValid="skip"/>
+    <Message value="Using the string for the component type variants results in additional overhead and is not safe. 
+    Suggestion is transform from .{0}(\&quot;{1}\&quot;) to .{0}&lt;{1}&gt;().{2}.">
+      <Argument>MethodName</Argument>
+      <Argument>StringLiteral</Argument>
+      <Argument>AvailableTypes.Length > 1 ? "\nThe type defined in several namespaces and behavior is unexpected" : string.Empty</Argument>
+    </Message> 
+    <Range>InvocationMethod.GetHighlightingRange()</Range>
+    <Behavour overlapResolvePolicy="WARNING" />
+  </Warning>
 
+  <Warning name="InvalidStringForType" configurableSeverity="Unity.InvalidStringForType">
+    <Parameter type="ILiteralExpression" name="argument" />
+    <Parameter type="string" name="typeLiteral" />
+    <Message value="{0} could not be found or any type with same name is unsuitable here."> 
```

Comment:
let's distinguish between this cases
and tell why it is unsuitable

Reply:
Also let's surrond the class name with quotes here, e.g.
`"Whatever" could not be found`